{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/anubrata/Documents/aarti_data/new_data'"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import os\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "os.getcwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "arti_data=pd.read_excel(\"all_samples_EDIT.xls\",sheet_name=0,header=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>stm_1_S13</th>\n",
       "      <th>C_Cip</th>\n",
       "      <th>A_cip</th>\n",
       "      <th>cip_2B_S14</th>\n",
       "      <th>ntrc_gDNA_2</th>\n",
       "      <th>SCV_2</th>\n",
       "      <th>Repl2_244M</th>\n",
       "      <th>Change_type</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Warnings</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Bio_type</th>\n",
       "      <th>Trancript_ID</th>\n",
       "      <th>Exon_ID</th>\n",
       "      <th>Exon_Rank</th>\n",
       "      <th>Effect</th>\n",
       "      <th>old_AA/new_AA</th>\n",
       "      <th>Old_codon/New_codon</th>\n",
       "      <th>Codon_Num(CDS)</th>\n",
       "      <th>Codon_Degeneracy</th>\n",
       "      <th>CDS_size</th>\n",
       "      <th>Codons_around</th>\n",
       "      <th>AAs_around</th>\n",
       "      <th>Custom_interval_ID</th>\n",
       "      <th>Unnamed: 28</th>\n",
       "      <th>Unnamed: 29</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>507</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4179.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>G/G</td>\n",
       "      <td>ggC/ggT</td>\n",
       "      <td>57</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4690.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>A/A</td>\n",
       "      <td>gcA/gcC</td>\n",
       "      <td>67</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>744</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>5187.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>G/G</td>\n",
       "      <td>ggA/ggC</td>\n",
       "      <td>136</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>747</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>5560.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>L/L</td>\n",
       "      <td>ctT/ctC</td>\n",
       "      <td>137</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>762</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>5371.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>G/G</td>\n",
       "      <td>ggG/ggA</td>\n",
       "      <td>142</td>\n",
       "      <td>4.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Position REF ALT stm_1_S13 C_Cip A_cip cip_2B_S14 ntrc_gDNA_2 SCV_2  \\\n",
       "0       507   C   T       ./.   1/1   1/1        1/1         ./.   ./.   \n",
       "1       537   A   C       ./.   1/1   1/1        1/1         ./.   ./.   \n",
       "2       744   A   C       ./.   1/1   1/1        1/1         ./.   ./.   \n",
       "3       747   T   C       ./.   1/1   1/1        1/1         ./.   ./.   \n",
       "4       762   G   A       ./.   1/1   1/1        1/1         ./.   ./.   \n",
       "\n",
       "  Repl2_244M Change_type  Coverage Warnings Gene_ID         Gene_name  \\\n",
       "0        1/1         SNP    4179.0    gene1    thrA  Transcript_gene1   \n",
       "1        1/1         SNP    4690.0    gene1    thrA  Transcript_gene1   \n",
       "2        1/1         SNP    5187.0    gene1    thrA  Transcript_gene1   \n",
       "3        1/1         SNP    5560.0    gene1    thrA  Transcript_gene1   \n",
       "4        1/1         SNP    5371.0    gene1    thrA  Transcript_gene1   \n",
       "\n",
       "                    Bio_type Trancript_ID            Exon_ID Exon_Rank  \\\n",
       "0  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       G/G   \n",
       "1  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       A/A   \n",
       "2  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       G/G   \n",
       "3  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       L/L   \n",
       "4  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       G/G   \n",
       "\n",
       "    Effect old_AA/new_AA  Old_codon/New_codon  Codon_Num(CDS)  \\\n",
       "0  ggC/ggT            57                  4.0             NaN   \n",
       "1  gcA/gcC            67                  4.0             NaN   \n",
       "2  ggA/ggC           136                  4.0             NaN   \n",
       "3  ctT/ctC           137                  4.0             NaN   \n",
       "4  ggG/ggA           142                  4.0             NaN   \n",
       "\n",
       "   Codon_Degeneracy  CDS_size  Codons_around  AAs_around  Custom_interval_ID  \\\n",
       "0               NaN       NaN            NaN         NaN                 NaN   \n",
       "1               NaN       NaN            NaN         NaN                 NaN   \n",
       "2               NaN       NaN            NaN         NaN                 NaN   \n",
       "3               NaN       NaN            NaN         NaN                 NaN   \n",
       "4               NaN       NaN            NaN         NaN                 NaN   \n",
       "\n",
       "   Unnamed: 28  Unnamed: 29  \n",
       "0          NaN          NaN  \n",
       "1          NaN          NaN  \n",
       "2          NaN          NaN  \n",
       "3          NaN          NaN  \n",
       "4          NaN          NaN  "
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.set_option('display.max_columns',None)\n",
    "pd.set_option('display.max_columns',None)\n",
    "#arti_data.drop(['CDS_size','Codons_around','Custom_interval_ID'],axis=1,inplace=True)\n",
    "#arti_data.drop(['Codon_Num(CDS)','AAs_around'],axis=1,inplace=True)\n",
    "#arti_data.drop(['Codon_Degeneracy','Unnamed: 28','Unnamed: 29'],axis=1,inplace=True)\n",
    "#arti_data['stm_1_S13']=arti_data['stm_1_S13'].replace('./.','Non_Var')\n",
    "#arti_data['ntrc_gDNA_2']=arti_data['ntrc_gDNA_2'].replace('./.','Non_Var')\n",
    "#arti_data['SCV_2']=arti_data['SCV_2'].replace('./.','Non_Var')\n",
    "arti_data.reset_index()\n",
    "arti_data.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/anubrata/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "pd.set_option('display.max_columns',None)\n",
    "pd.set_option('display.max_rows',None)\n",
    "#arti_data.groupby('Bio_type')['Trancript_ID'].count()\n",
    "arti_data.groupby('Trancript_ID')['Trancript_ID'].count().sort_values(ascending=True).to_csv(\"error_Trancript_ID.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/anubrata/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:4: FutureWarning: The signature of `Series.to_csv` was aligned to that of `DataFrame.to_csv`, and argument 'header' will change its default value from False to True: please pass an explicit value to suppress this warning.\n",
      "  after removing the cwd from sys.path.\n"
     ]
    }
   ],
   "source": [
    "pd.set_option('display.max_columns',None)\n",
    "pd.set_option('display.max_rows',None)\n",
    "#arti_data.groupby('Bio_type')['Trancript_ID'].count()\n",
    "arti_data.groupby('Bio_type')['Bio_type'].count().sort_values(ascending=False).to_csv(\"error_Bio_type.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>stm_1_S13</th>\n",
       "      <th>C_Cip</th>\n",
       "      <th>A_cip</th>\n",
       "      <th>cip_2B_S14</th>\n",
       "      <th>ntrc_gDNA_2</th>\n",
       "      <th>SCV_2</th>\n",
       "      <th>Repl2_244M</th>\n",
       "      <th>Change_type</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Warnings</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Bio_type</th>\n",
       "      <th>Trancript_ID</th>\n",
       "      <th>Exon_ID</th>\n",
       "      <th>Exon_Rank</th>\n",
       "      <th>Effect</th>\n",
       "      <th>old_AA/new_AA</th>\n",
       "      <th>Old_codon/New_codon</th>\n",
       "      <th>Codon_Num(CDS)</th>\n",
       "      <th>Codon_Degeneracy</th>\n",
       "      <th>CDS_size</th>\n",
       "      <th>Codons_around</th>\n",
       "      <th>AAs_around</th>\n",
       "      <th>Custom_interval_ID</th>\n",
       "      <th>Unnamed: 28</th>\n",
       "      <th>Unnamed: 29</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>42</td>\n",
       "      <td>5890</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4610.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>62</td>\n",
       "      <td>7420</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>3932.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>63</td>\n",
       "      <td>7432</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4060.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>64</td>\n",
       "      <td>7508</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4493.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>78</td>\n",
       "      <td>9326</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4869.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43278</td>\n",
       "      <td>4856558</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>1961.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43279</td>\n",
       "      <td>4856655</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>3595.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43290</td>\n",
       "      <td>4857443</td>\n",
       "      <td>A</td>\n",
       "      <td>AAAT</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43291</td>\n",
       "      <td>4857445</td>\n",
       "      <td>A</td>\n",
       "      <td>ATAATAAGAGAT,ATAATAAGAGATTACGTC</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>2/2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>43292</td>\n",
       "      <td>3</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>559.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8892 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Position REF                              ALT stm_1_S13 C_Cip A_cip  \\\n",
       "42         5890   T                                C       ./.   1/1   1/1   \n",
       "62         7420   G                                T       ./.   1/1   1/1   \n",
       "63         7432   A                                G       ./.   1/1   1/1   \n",
       "64         7508   T                                C       ./.   1/1   1/1   \n",
       "78         9326   A                                G       ./.   1/1   1/1   \n",
       "...         ...  ..                              ...       ...   ...   ...   \n",
       "43278   4856558   T                                C       ./.   1/1   1/1   \n",
       "43279   4856655   A                                C       ./.   1/1   1/1   \n",
       "43290   4857443   A                             AAAT       ./.   ./.   1/1   \n",
       "43291   4857445   A  ATAATAAGAGAT,ATAATAAGAGATTACGTC       ./.   ./.   ./.   \n",
       "43292         3   G                                A       ./.   ./.   ./.   \n",
       "\n",
       "      cip_2B_S14 ntrc_gDNA_2 SCV_2 Repl2_244M Change_type  Coverage  \\\n",
       "42           1/1         ./.   ./.        1/1         SNP    4610.0   \n",
       "62           1/1         ./.   ./.        1/1         SNP    3932.0   \n",
       "63           1/1         ./.   ./.        1/1         SNP    4060.0   \n",
       "64           1/1         ./.   ./.        1/1         SNP    4493.0   \n",
       "78           1/1         ./.   ./.        1/1         SNP    4869.0   \n",
       "...          ...         ...   ...        ...         ...       ...   \n",
       "43278        1/1         ./.   ./.        1/1         SNP    1961.0   \n",
       "43279        1/1         ./.   ./.        1/1         SNP    3595.0   \n",
       "43290        ./.         ./.   ./.        ./.         NaN       NaN   \n",
       "43291        ./.         1/1   ./.        2/2         NaN       NaN   \n",
       "43292        ./.         1/1   ./.        ./.         SNP     559.0   \n",
       "\n",
       "         Warnings Gene_ID Gene_name Bio_type Trancript_ID Exon_ID Exon_Rank  \\\n",
       "42     INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "62     INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "63     INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "64     INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "78     INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "...           ...     ...       ...      ...          ...     ...       ...   \n",
       "43278  INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "43279  INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "43290         NaN     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "43291         NaN     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "43292  INTERGENIC     NaN       NaN      NaN          NaN     NaN       NaN   \n",
       "\n",
       "      Effect old_AA/new_AA  Old_codon/New_codon  Codon_Num(CDS)  \\\n",
       "42       NaN           NaN                  NaN             NaN   \n",
       "62       NaN           NaN                  NaN             NaN   \n",
       "63       NaN           NaN                  NaN             NaN   \n",
       "64       NaN           NaN                  NaN             NaN   \n",
       "78       NaN           NaN                  NaN             NaN   \n",
       "...      ...           ...                  ...             ...   \n",
       "43278    NaN           NaN                  NaN             NaN   \n",
       "43279    NaN           NaN                  NaN             NaN   \n",
       "43290    NaN           NaN                  NaN             NaN   \n",
       "43291    NaN           NaN                  NaN             NaN   \n",
       "43292    NaN           NaN                  NaN             NaN   \n",
       "\n",
       "       Codon_Degeneracy  CDS_size  Codons_around  AAs_around  \\\n",
       "42                  NaN       NaN            NaN         NaN   \n",
       "62                  NaN       NaN            NaN         NaN   \n",
       "63                  NaN       NaN            NaN         NaN   \n",
       "64                  NaN       NaN            NaN         NaN   \n",
       "78                  NaN       NaN            NaN         NaN   \n",
       "...                 ...       ...            ...         ...   \n",
       "43278               NaN       NaN            NaN         NaN   \n",
       "43279               NaN       NaN            NaN         NaN   \n",
       "43290               NaN       NaN            NaN         NaN   \n",
       "43291               NaN       NaN            NaN         NaN   \n",
       "43292               NaN       NaN            NaN         NaN   \n",
       "\n",
       "       Custom_interval_ID  Unnamed: 28  Unnamed: 29  \n",
       "42                    NaN          NaN          NaN  \n",
       "62                    NaN          NaN          NaN  \n",
       "63                    NaN          NaN          NaN  \n",
       "64                    NaN          NaN          NaN  \n",
       "78                    NaN          NaN          NaN  \n",
       "...                   ...          ...          ...  \n",
       "43278                 NaN          NaN          NaN  \n",
       "43279                 NaN          NaN          NaN  \n",
       "43290                 NaN          NaN          NaN  \n",
       "43291                 NaN          NaN          NaN  \n",
       "43292                 NaN          NaN          NaN  \n",
       "\n",
       "[8892 rows x 30 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data[arti_data.Trancript_ID!=1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(43294, 30)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>stm_1_S13</th>\n",
       "      <th>C_Cip</th>\n",
       "      <th>A_cip</th>\n",
       "      <th>cip_2B_S14</th>\n",
       "      <th>ntrc_gDNA_2</th>\n",
       "      <th>SCV_2</th>\n",
       "      <th>Repl2_244M</th>\n",
       "      <th>Change_type</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Warnings</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Bio_type</th>\n",
       "      <th>Trancript_ID</th>\n",
       "      <th>Exon_ID</th>\n",
       "      <th>Exon_Rank</th>\n",
       "      <th>Effect</th>\n",
       "      <th>old_AA/new_AA</th>\n",
       "      <th>Old_codon/New_codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>405</td>\n",
       "      <td>39536</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>515.0</td>\n",
       "      <td>INTERGENIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2827</td>\n",
       "      <td>294865</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>907.0</td>\n",
       "      <td>WARNING_TRANSCRIPT_INCOMPLETE</td>\n",
       "      <td>gene253</td>\n",
       "      <td>aspU</td>\n",
       "      <td>rna5</td>\n",
       "      <td>id273</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>V/V</td>\n",
       "      <td>gtC/gtT</td>\n",
       "      <td>9.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3046</td>\n",
       "      <td>314224</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>916.0</td>\n",
       "      <td>gene272</td>\n",
       "      <td>STM0273</td>\n",
       "      <td>Transcript_gene272</td>\n",
       "      <td>Exon_NC_003197.2_313933_314475</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>V/V</td>\n",
       "      <td>gtC/gtT</td>\n",
       "      <td>97</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4007</td>\n",
       "      <td>416667</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>895.0</td>\n",
       "      <td>gene366</td>\n",
       "      <td>prpR</td>\n",
       "      <td>Transcript_gene366</td>\n",
       "      <td>Exon_NC_003197.2_416087_417712</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>K/K</td>\n",
       "      <td>aaA/aaG</td>\n",
       "      <td>349</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4080</td>\n",
       "      <td>422380</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>./.</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>./.</td>\n",
       "      <td>SNP</td>\n",
       "      <td>702.0</td>\n",
       "      <td>gene370</td>\n",
       "      <td>prpE</td>\n",
       "      <td>Transcript_gene370</td>\n",
       "      <td>Exon_NC_003197.2_421688_423574</td>\n",
       "      <td>1</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "      <td>S/F</td>\n",
       "      <td>tCc/tTc</td>\n",
       "      <td>231</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Position REF ALT stm_1_S13 C_Cip A_cip cip_2B_S14 ntrc_gDNA_2 SCV_2  \\\n",
       "405      39536   C   T   Non_Var   ./.   ./.        ./.     Non_Var   1/1   \n",
       "2827    294865   C   T   Non_Var   ./.   ./.        ./.     Non_Var   1/1   \n",
       "3046    314224   C   T   Non_Var   ./.   ./.        ./.     Non_Var   1/1   \n",
       "4007    416667   T   C   Non_Var   ./.   ./.        ./.     Non_Var   1/1   \n",
       "4080    422380   C   T   Non_Var   ./.   ./.        ./.     Non_Var   1/1   \n",
       "\n",
       "     Repl2_244M Change_type  Coverage                       Warnings  Gene_ID  \\\n",
       "405         ./.         SNP     515.0                     INTERGENIC      NaN   \n",
       "2827        ./.         SNP     907.0  WARNING_TRANSCRIPT_INCOMPLETE  gene253   \n",
       "3046        ./.         SNP     916.0                        gene272  STM0273   \n",
       "4007        ./.         SNP     895.0                        gene366     prpR   \n",
       "4080        ./.         SNP     702.0                        gene370     prpE   \n",
       "\n",
       "               Gene_name                        Bio_type Trancript_ID  \\\n",
       "405                  NaN                             NaN          NaN   \n",
       "2827                aspU                            rna5        id273   \n",
       "3046  Transcript_gene272  Exon_NC_003197.2_313933_314475            1   \n",
       "4007  Transcript_gene366  Exon_NC_003197.2_416087_417712            1   \n",
       "4080  Transcript_gene370  Exon_NC_003197.2_421688_423574            1   \n",
       "\n",
       "                    Exon_ID          Exon_Rank   Effect old_AA/new_AA  \\\n",
       "405                     NaN                NaN      NaN           NaN   \n",
       "2827                      1  SYNONYMOUS_CODING      V/V       gtC/gtT   \n",
       "3046      SYNONYMOUS_CODING                V/V  gtC/gtT            97   \n",
       "4007      SYNONYMOUS_CODING                K/K  aaA/aaG           349   \n",
       "4080  NON_SYNONYMOUS_CODING                S/F  tCc/tTc           231   \n",
       "\n",
       "      Old_codon/New_codon  \n",
       "405                   NaN  \n",
       "2827                  9.0  \n",
       "3046                  4.0  \n",
       "4007                  2.0  \n",
       "4080                  1.0  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data[arti_data.SCV_2!='Non_Var'].head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene_ID        Warnings     \n",
       "Gene_gene2690  Gene_gene2690    179\n",
       "STM4261        gene4264         174\n",
       "metH           gene4191          73\n",
       "STM4489        gene4493          72\n",
       "carB           gene66            59\n",
       "                               ... \n",
       "ymdA           gene1142           1\n",
       "ppa            gene4418           1\n",
       "yqjB           gene3227           1\n",
       "yceI           gene1153           1\n",
       "Gene_gene1182  Gene_gene1182      1\n",
       "Name: Gene_ID, Length: 3879, dtype: int64"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data.groupby(['Gene_ID','Warnings'])['Gene_ID'].count().sort_values(ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Change_type\n",
       "DEL        4\n",
       "INS       33\n",
       "SNP    42952\n",
       "Name: Change_type, dtype: int64"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data.groupby('Change_type')['Change_type'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene_ID        Change_type\n",
       "Gene_gene1182  SNP             1\n",
       "Gene_gene1241  SNP             1\n",
       "Gene_gene1244  SNP             2\n",
       "Gene_gene1470  SNP            10\n",
       "Gene_gene1549  SNP            13\n",
       "                              ..\n",
       "zntR           SNP             6\n",
       "znuB           SNP             3\n",
       "zraP           SNP             3\n",
       "zur            SNP             6\n",
       "zwf            SNP            14\n",
       "Name: Change_type, Length: 3887, dtype: int64"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data.groupby(['Gene_ID','Change_type'])['Change_type'].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Position</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>stm_1_S13</th>\n",
       "      <th>C_Cip</th>\n",
       "      <th>A_cip</th>\n",
       "      <th>cip_2B_S14</th>\n",
       "      <th>ntrc_gDNA_2</th>\n",
       "      <th>SCV_2</th>\n",
       "      <th>Repl2_244M</th>\n",
       "      <th>Change_type</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Warnings</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Gene_name</th>\n",
       "      <th>Bio_type</th>\n",
       "      <th>Trancript_ID</th>\n",
       "      <th>Exon_ID</th>\n",
       "      <th>Exon_Rank</th>\n",
       "      <th>Effect</th>\n",
       "      <th>old_AA/new_AA</th>\n",
       "      <th>Old_codon/New_codon</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>507</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4179.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>G/G</td>\n",
       "      <td>ggC/ggT</td>\n",
       "      <td>57</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>537</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>1/1</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>Non_Var</td>\n",
       "      <td>1/1</td>\n",
       "      <td>SNP</td>\n",
       "      <td>4690.0</td>\n",
       "      <td>gene1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>A/A</td>\n",
       "      <td>gcA/gcC</td>\n",
       "      <td>67</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Position REF ALT stm_1_S13 C_Cip A_cip cip_2B_S14 ntrc_gDNA_2    SCV_2  \\\n",
       "0       507   C   T   Non_Var   1/1   1/1        1/1     Non_Var  Non_Var   \n",
       "1       537   A   C   Non_Var   1/1   1/1        1/1     Non_Var  Non_Var   \n",
       "\n",
       "  Repl2_244M Change_type  Coverage Warnings Gene_ID         Gene_name  \\\n",
       "0        1/1         SNP    4179.0    gene1    thrA  Transcript_gene1   \n",
       "1        1/1         SNP    4690.0    gene1    thrA  Transcript_gene1   \n",
       "\n",
       "                    Bio_type Trancript_ID            Exon_ID Exon_Rank  \\\n",
       "0  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       G/G   \n",
       "1  Exon_NC_003197.2_336_2798            1  SYNONYMOUS_CODING       A/A   \n",
       "\n",
       "    Effect old_AA/new_AA  Old_codon/New_codon  \n",
       "0  ggC/ggT            57                  4.0  \n",
       "1  gcA/gcC            67                  4.0  "
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arti_data.head(n=2)\n",
    "#rev_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "data2=pd.read_csv('A_cip-1_New.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Position</th>\n",
       "      <th>Reference</th>\n",
       "      <th>Variant</th>\n",
       "      <th>Type</th>\n",
       "      <th>Zygosity</th>\n",
       "      <th>Quality</th>\n",
       "      <th>Coverage</th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Trancript_ID</th>\n",
       "      <th>Exon_ID</th>\n",
       "      <th>Exon_Rank</th>\n",
       "      <th>Effect</th>\n",
       "      <th>old_AA/new_AA</th>\n",
       "      <th>Old_codon/New_codon</th>\n",
       "      <th>Codon_Num(CDS)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>507</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Hom</td>\n",
       "      <td>221.999</td>\n",
       "      <td>1028</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>G/G</td>\n",
       "      <td>ggC/ggT</td>\n",
       "      <td>57.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>NC_003197.2</td>\n",
       "      <td>537</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>SNP</td>\n",
       "      <td>Hom</td>\n",
       "      <td>221.999</td>\n",
       "      <td>1130</td>\n",
       "      <td>thrA</td>\n",
       "      <td>Transcript_gene1</td>\n",
       "      <td>Exon_NC_003197.2_336_2798</td>\n",
       "      <td>1</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "      <td>A/A</td>\n",
       "      <td>gcA/gcC</td>\n",
       "      <td>67.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chrom  Position Reference Variant Type Zygosity  Quality  Coverage  \\\n",
       "0  NC_003197.2       507         C       T  SNP      Hom  221.999      1028   \n",
       "1  NC_003197.2       537         A       C  SNP      Hom  221.999      1130   \n",
       "\n",
       "  Gene_ID      Trancript_ID                    Exon_ID Exon_Rank  \\\n",
       "0    thrA  Transcript_gene1  Exon_NC_003197.2_336_2798         1   \n",
       "1    thrA  Transcript_gene1  Exon_NC_003197.2_336_2798         1   \n",
       "\n",
       "              Effect old_AA/new_AA Old_codon/New_codon  Codon_Num(CDS)  \n",
       "0  SYNONYMOUS_CODING           G/G             ggC/ggT            57.0  \n",
       "1  SYNONYMOUS_CODING           A/A             gcA/gcC            67.0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data2.head(n=2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene_ID</th>\n",
       "      <th>Position</th>\n",
       "      <th>Effect</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <td>0</td>\n",
       "      <td>thrA</td>\n",
       "      <td>507</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>1</td>\n",
       "      <td>thrA</td>\n",
       "      <td>537</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>2</td>\n",
       "      <td>thrA</td>\n",
       "      <td>744</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>3</td>\n",
       "      <td>thrA</td>\n",
       "      <td>747</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>4</td>\n",
       "      <td>thrA</td>\n",
       "      <td>762</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>5</td>\n",
       "      <td>thrA</td>\n",
       "      <td>828</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>6</td>\n",
       "      <td>thrA</td>\n",
       "      <td>831</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>7</td>\n",
       "      <td>thrA</td>\n",
       "      <td>877</td>\n",
       "      <td>NON_SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>8</td>\n",
       "      <td>thrA</td>\n",
       "      <td>1245</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <td>9</td>\n",
       "      <td>thrA</td>\n",
       "      <td>1254</td>\n",
       "      <td>SYNONYMOUS_CODING</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Gene_ID  Position                 Effect\n",
       "0    thrA       507      SYNONYMOUS_CODING\n",
       "1    thrA       537      SYNONYMOUS_CODING\n",
       "2    thrA       744      SYNONYMOUS_CODING\n",
       "3    thrA       747      SYNONYMOUS_CODING\n",
       "4    thrA       762      SYNONYMOUS_CODING\n",
       "5    thrA       828      SYNONYMOUS_CODING\n",
       "6    thrA       831      SYNONYMOUS_CODING\n",
       "7    thrA       877  NON_SYNONYMOUS_CODING\n",
       "8    thrA      1245      SYNONYMOUS_CODING\n",
       "9    thrA      1254      SYNONYMOUS_CODING"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data=data2.loc[:,['Gene_ID','Position','Effect']]\n",
    "new_data.head(n=10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(40858, 43294)"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_data),len(arti_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "old_data=pd.read_csv('Cip_2B_S14.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(40858, 43294, 42668)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(new_data),len(arti_data),len(old_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "comp_data=arti_data.merge(new_data,how='outer',on='Position')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "comp_data1=pd.merge(arti_data,new_data,how='outer',on='Position')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "any(comp_data!=comp_data1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
